examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/XdsHelloWorldClient.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.examples.helloworldxds;
+
+import io.grpc.Channel;
+import io.grpc.ChannelCredentials;
+import io.grpc.Grpc;
+import io.grpc.InsecureChannelCredentials;
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.xds.XdsChannelCredentials;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A simple xDS client that requests a greeting from {@code HelloWorldServer} or {@link
+ * XdsHelloWorldServer}.
+ */
+public class XdsHelloWorldClient {
+  private static final Logger logger = Logger.getLogger(XdsHelloWorldClient.class.getName());
+
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  /** Construct client for accessing HelloWorld server using the existing channel. */
+  public XdsHelloWorldClient(Channel channel) {
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+  }
+
+  /** Say hello to server. */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+    logger.info("Greeting: " + response.getMessage());
+  }
+
+  /**
+   * Greet server. If provided, the first element of {@code args} is the name to use in the
+   * greeting. The second argument is the target server. A {@code --secure} flag is also accepted.
+   */
+  public static void main(String[] args) throws Exception {
+    String user = "xds world";
+    // The example defaults to the same behavior as the hello world example. To enable xDS, pass an
+    // "xds:"-prefixed string as the target.
+    String target = "localhost:50051";
```

Comment:
Is this a good default for this example? When `--secure` is used and target is omitted, the `--secure` flag has no meaning. Also when the target doesn't start with `xds:///` also the flag has no meaning.

Reply:
> Is this a good default for this example?

If there was a better default (like an xds: string), I'd consider it, but I don't think there is a better default. But I think it really shows that "the only thing that makes this xds is the target string." This is the current example behavior, and I think that is appropriate to keep.

> When --secure is used and target is omitted, the --secure flag has no meaning.

Not really any different than its normal meaning. It has the meaning "if xds provides security details use them, otherwise use the fallback." And here xds doesn't provide any security details.

> Also when the target doesn't start with xds:/// also the flag has no meaning.

That was already the case. Why bring that up?

Reply:
> If there was a better default (like an xds: string), I'd consider it, but I don't think there is a better default. But I think it really shows that "the only thing that makes this xds is the target string." This is the current example behavior, and I think that is appropriate to keep.
> 
> > When --secure is used and target is omitted, the --secure flag has no meaning.
> 
> Not really any different than its normal meaning. It has the meaning "if xds provides security details use them, otherwise use the fallback." And here xds doesn't provide any security details.
> 
> > Also when the target doesn't start with xds:/// also the flag has no meaning.
> 
> That was already the case. Why bring that up?

In theory the code can do these checks and issue warnings (or exit). Not suggesting we do it just for an example.

Reply:
Ah, okay. I don't think we would want to issue a warning. Maybe just a "notice." Using the xds credentials with a non-xds target isn't _wrong_; we purposefully allow it. But I do think that's a bit much for the example. The argument parsing is already an unfortunate amount of logic. And I don't want to encouraging any sort of "hey! let's parse this target string!" behavior from users. Instead, a comment is probably the cleaner approach, and I think one covers that detail already.